restore.py

```diff
@@ -192,6 +192,11 @@ def restore_partitions():
             util.assertDir(os.path.dirname(dst_file))
             boot_config.commit(dst_file)
 
+            # prepare for boot loader restoration, before unmounting backup_fs
+            if boot_config.src_fmt == 'grub2' and efi_boot:
+                branding = util.readKeyValueFile(os.path.join(backup_fs.mount_point, constants.INVENTORY_FILE))
+                branding['product-brand'] = branding['PRODUCT_BRAND']
+
             # repartition if needed
             backup_fs.unmount()
```

Comment:
Maybe just move this line inside the below `if` ?
We are restoring MBR so we won't have EFI to restore.

Reply:
possibly, but that rather looks like some improvement that can wait for later, here I'm solely focused on unbreaking the nominal case